src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,53 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+/**
+ * <h1>Deadline tasks with date</h1>
+ * The Deadline class is a specific subclass of the Task that allows user to
+ * add a stipulated end date for their keyed input. By default, it assumes
+ * the Deadline task is undone, because it does not make sense for a user to
+ * enter a new Deadline task if it is done. However, should the user want to
+ * input a past done Deadline task, he is allowed to do so. The user can
+ * also toggle the status of the Deadline task as done or undone. Lastly,
+ * the Deadline date can be both a day, and a local date.
+ * @author Muhammad Reyaaz
+ * @version %I% %G%
+ * @since 11
+ *
+ */
+
+class Deadline extends Task {
+    protected String by;
```

Comment:
I think it would be better if you use a more intuitive name. In addition, is there any reason why you store the deadline as String instead of LocalDate, and then convert it to LocalDate later with the method localDateParser()?
```suggestion
    protected String endDate;
```